R/human_infection.R

```diff
@@ -400,14 +409,44 @@ schedule_infections <- function(
       to_infect
     )
   }
-
-  if(to_infect_asym$size() > 0) {
-    update_to_asymptomatic_infection(
-      variables,
-      parameters,
-      timestep,
-      to_infect_asym
-    )
+  
+  # falciparum infection can result in D or A
+  if(parameters$parasite == "falciparum"){
+    to_infect_asym <- clinical_infections$copy()$not(FALSE)$and(infections)$and(included)
```

Comment:
If $not is returning a copy you do not need to copy beforehand: 

```suggestion
    to_infect_asym <- clinical_infections$not(FALSE)$and(infections)$and(included)
```

Reply:
I've played around with these a bit to make sure the outputs are being generated, and I think it's working now, including the corrections you suggested.